File: java/algorithm/sort/BubbleSort.java
package algorithm.sort;

/**
 * @author roseduan[roseduan520@gmail.com]
 * @time 2019/12/7 15:25
 * @description 冒泡排序
 */
public class BubbleSort {

    public static void sort1(int[] data){
        for (int i = 0; i < data.length; i++) {
            for (int j = 0; j < data.length - i - 1; j++) {
                if (data[j] > data[j + 1]){
                    int k = data[j];
                    data[j] = data[j + 1];
                    data[j + 1] = k;
                }
            }
        }
    }

    public static void sort2(int[] data){
        boolean swap = true;
        for (int i = 0; i < data.length && swap; i++) {
            swap = false;
            for (int j = 0; j < data.length - i - 1; j++) {
                if (data[j] > data[j + 1]) {
                    int k = data[j];
                    data[j] = data[j + 1];
                    data[j + 1] = k;
                    swap = true;
                }
            }
        }
    }

}

File: cpp/sort/bubble_sort.cpp
//
// Created by roseduan on 2021/9/21.
//
#include "vector"
#include "iostream"
using namespace std;

void BubbleSort(vector<int>& data) {
    for (int i = 0; i < data.size() - 1; ++i) {
        for (int j = i + 1; j < data.size(); ++j) {
            if (data[i] > data[j]) {
                int temp = data[i];
                data[i] = data[j];
                data[j] = temp;
            }
        }
    }
}

int main() {
    vector<int> data {6};
    BubbleSort(data);

    for (int i : data) {
        cout << i << endl;
    }
}
